pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -108,10 +110,12 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(
 					// This is a HTTPS server, where we are doing TLS termination. Build a http connection manager with TLS context
 					routeName := mergedGateway.RouteNamesByServer[server]
 					filterChainOpts = append(filterChainOpts, configgen.createGatewayHTTPFilterChainOpts(node, server, routeName, push.Mesh.SdsUdsPath))
+					filterChains = append(filterChains, plugin.FilterChain{ListenerProtocol: plugin.ListenerProtocolHTTP})
 				} else {
 					// passthrough or tcp, yields multiple filter chains
 					filterChainOpts = append(filterChainOpts, configgen.createGatewayTCPFilterChainOpts(node, push,
 						server, map[string]bool{mergedGateway.GatewayNameForServer[server]: true})...)
```

Comment:
Actually `createGatewayTCPFilterChainOpts` return value is of array type. It is confusing as you do this below, so could you also change that function return value?

Reply:
@hzxuzhonghu Thank you for your review, i'm sorry to forget change the return value of the createGatewayTCPFilterChainOpts.

Reply:
@hzxuzhonghu It's looks the function createGatewayTCPFilterChainOpts has some special logic to return array values when case of passthrough server to return buildGatewayNetworkFiltersFromTLSRoutes function, so i think we can handle array to what we needs.